packages/admin/cms-admin/src/blocks/seo/useSeoTagGeneration.tsx

```diff
@@ -0,0 +1,125 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { useErrorDialog } from "@comet/admin";
+import { useCallback } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { useDocumentContentGenerationApi } from "../../documents/DocumentContentGenerationContext";
+import { GQLGenerateSeoTagsMutation, GQLGenerateSeoTagsMutationVariables } from "./useSeoTagGeneration.generated";
+
+let seoTagsCache: GQLGenerateSeoTagsMutation["generateSeoTags"] & { content: string } = {
+    content: "",
+    htmlTitle: "",
+    metaDescription: "",
+    openGraphTitle: "",
+    openGraphDescription: "",
+};
+
+let pendingRequest: { content: string; request: Promise<GQLGenerateSeoTagsMutation["generateSeoTags"]> } | undefined;
+
+export function useSeoTagGeneration() {
+    const documentContentGenerationApi = useDocumentContentGenerationApi();
+    const errorDialog = useErrorDialog();
+    const apolloClient = useApolloClient();
+
+    const getSeoTag = useCallback(
```

Comment:
Naming: this should be getSeoTag**s**

Reply:
48bc294e9d6d3cc0e1d8197be11c4d366aed933d